spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/InternalRowWrapper.java

```diff
@@ -71,8 +77,12 @@ public <T> void set(int pos, T value) {
     row.update(pos, value);
   }
 
-  private static BiFunction<InternalRow, Integer, ?> getter(DataType type) {
+  private static BiFunction<InternalRow, Integer, ?> getter(Type icebergType, DataType type) {
     if (type instanceof StringType) {
+      if (Type.TypeID.UUID == icebergType.typeId()) {
```

Comment:
can we add a comment here explaining that a StringType may be a UUID-type hence return UUID also should we explicity check the else case saying if it's not StringType either throw UnReachable or Unsupported ops or null accordingly ?